ProcessOut/Classes/ProcessOut.swift

```diff
@@ -529,6 +547,28 @@ public class ProcessOut {
         })
     }
     
+    /// Generates a query parameter string to facilitate pagination on many endpoints.
+    /// For more information, see https://docs.processout.com/refs/#pagination
+    ///
+    /// - Parameter paginationOptions: Pagination options to use
+    /// - Returns: An empty string or a string containing a set of query parameters.
+    /// Note that the returned string is not prefixed or suffixed with ? or &, so you may need to do this yourself depending on where these parameters will appear in your URL
+    private static func generatePaginationParamsString(paginationOptions: PaginationOptions) -> String {
+        // Construct the individual query params and store them in an array
+        let paginationParams: [String?] = [
+            paginationOptions.StartAfter != nil ? "start_after=" + paginationOptions.StartAfter! : nil,
+            paginationOptions.EndBefore != nil ? "end_before=" + paginationOptions.EndBefore! : nil,
+            paginationOptions.Limit != nil ? "limit=" + String(paginationOptions.Limit!) : nil,
+            paginationOptions.Order != nil ? "order=" + paginationOptions.Order! : nil
+        ]
+
+        // Remove any nil values from the array
+        let filteredPaginationParams = paginationParams.flatMap {$0}
```

Comment:
Thinking out-loud... What do you think to moving this into a method `ToStringArray()` and putting that on the `PaginationOptions` struct? It is leaking a little bit of implementation detail ... but ... would mean that if you ever needed to add another parameter option, you'd only need to do it in one place (if that makes sense).

I'd suggest leaving the return below where it is, and add `PaginationOptions.ToStringArray()` to this method.

Reply:
Thank you for your feedback 🙂 

I considered abstracting some or all of the functionality in the `generatePaginationParamsString()` method into the `PaginationOptions` struct, but I ultimately decided against it because, as you mentioned, doing so would inadvertently expose extraneous functionality to end-users who will almost definitely never use it.

Unless it is significantly refactored to generate each element in the array programatically, adding another parameter will still require it to be manually added to the `paginationParams` array. With this in mind, I personally don't think that having it closer to the `PaginationOptions` struct in the `ProcessOut.swift` file would offer enough benefit in terms of maintainability to make it worth being unnecessarily publicly accessible. In addition, the string array is very unlikely to have any application outside of this `generatePaginationParamsString()` method, so I don't think that moving it into the `PaginationOptions` struct would offer many opportunities to maximise code reuse.